[Tham Jin Lin] iP

jltham opened the pull request:
# **Jin Lin's Duke**
> "Nuke Duke Juked Puke Luke" - How Now Brown Cow, 2022 ([sauce](https://www.youtube.com/watch?v=Yry66pkQx-E))


My duke helps you be lazy and lie in bed all day :sleeping_bed: 

It will ~~most definitely~~ :crossed_fingers:
- Greet you
- Tell you when you're inadequate
- Bid you farewell

Let me give you a step by step tutorial to it :zany_face:
1. Say hi to the machine
2. Tell Duke your wishes
3. That's all !

Here, I'll give you a sneak peek to the most complex code, the `goat` method ! :face_with_spiral_eyes:
``` java
public static void goat() {
    print("Hello World!");
}
```

Alright that's enough giving from me ! Follow me for more content !
- [ ] MSN
- [x] MySpace
- [ ] Skype

Commit:
Add Gradle support

Commit:
added level-1 requirements

Commit:
added level-2 requirements

Commit:
Added Task class and mark/unmark functions

Commit:
added different specifications of tasks

Commit:
added DukeException and exception handling

Commit:
added delete function

Commit:
Edited Duke

Commit:
Added automated Text UI Testing

michaelseyo commented on src/main/java/Deadlines.java:
```diff
@@ -0,0 +1,23 @@
+public class Deadlines extends Task {
```
Perhaps the class name could be singular? 
I like that you used inheritance for different types of Task.

michaelseyo commented on src/main/java/Deadlines.java:
```diff
@@ -0,0 +1,23 @@
+public class Deadlines extends Task {
+    protected static String type = "[D]";
```
Would it be better for the type to be static final and capitalized as in the coding standards for constants since this type won't be changing?

DaneMarc replied:
Agreed

michaelseyo commented on src/main/java/DukeTest.java:
```diff
@@ -0,0 +1,108 @@
+
+            exit();
+            scanner.close();
+        } catch (DukeException e) {
```
I like that you considered the many potential errors in the user inputs 😄

michaelseyo commented on src/main/java/DukeTest.java:
```diff
@@ -0,0 +1,108 @@
+
+            while (!(isExit(command))) {
+                switch (command) {
+                    case "list":
```
Just a reminder that the case could be indented in the same line as the switch command as in the coding standards

michaelseyo commented on src/main/java/DukeTest.java:
```diff
@@ -0,0 +1,108 @@
+            String command = splitInput[0];
+
+            while (!(isExit(command))) {
+                switch (command) {
```
Great that you used switch case and have modularized your code to perform the respective actions for the different conditions, looks neat!

michaelseyo reviewed (commented):
Generally looks good except just some naming to be resolved.

DaneMarc commented on src/main/java/Deadlines.java:
```diff
@@ -0,0 +1,23 @@
+
+    @Override
+    public String toString() {
+        return this.isDone ? "[D][X] " + this.description
```
Perhaps you could remove the 'this' keywords since the variables are not shadowed

DaneMarc commented on src/main/java/DukeTest.java:
```diff
@@ -0,0 +1,108 @@
+    private static void addTask(Task task) {
+        taskList.add(task);
+        System.out.println("Got it. I've added this task:\n" + task.toString() +
+                "\nNow you have " + taskList.size() + " tasks in the list.");
```
Proper indentation 👍

DaneMarc reviewed (commented):
Looks good other than a few little details that should be easy to fix. Comments are missing.

Commit:
Implemented new Save function

Commit:
Stored deadline date and times as LocalDate and LocalTime objects

Commit:
Merge branch 'branch-Level-8'

Commit:
Made the code more OOP

Commit:
Packaged files

Commit:
Packaged the App as a JAR file

Commit:
Added JavaDoc

Commit:
Revert "Added JavaDoc"

This reverts commit 560f9523d16a56b27ce0260ea6b80eea53545525.

Commit:
Revert "Revert "Added JavaDoc""

This reverts commit 4b692d2f21ac24bd643b1f59c3afa7521e9d0290.

Commit:
Updated JavaDocs

Commit:
Updated Coding Standards

Commit:
Implemented Find function

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Merge branch 'branch-Level-9'

Commit:
Merge branch 'add-gradle-support' of https://github.com/jltham/ip

Commit:
Added gradle

Commit:
Added some trivial JUnit tests

Commit:
Added Checkstyle that works with Gradle

Commit:
Added basic GUI

Commit:
Edited Ui to follow GUI implementations.

Commit:
Merge branch 'branch-Level-10'

Commit:
Added assertions throughout the code base.

Important assumptions mainly regarding validity of parameters exist in
various methods of the different classes in Duke.

Certain assumptions may not hold and that can cause the program to fail.

Let's add assertions at every part of the code where important
assumptions are being held to ensure that they hold true.

Using assertions to document assumptions within the program allows the
runtime to verify them. Upon assertion failure, the program typically
terminates with an error message and this prevents possible bugs in the
program from harming the program.

Commit:
Amended some of code of bad quality.

Commit:
Merge pull request #1 from jltham/branch-A-Assertions

Branch A-Assertions

Commit:
Merge branch 'master' of https://github.com/jltham/ip

* 'master' of https://github.com/jltham/ip:
  Added assertions throughout the code base.

Commit:
Merge branch 'master' into branch-A-CodeQuality

Commit:
Merge branch 'branch-A-CodeQuality'

Commit:
Added .yml file for CI.

Commit:
Created new classes: ChangeMark to contain both Mark and Unmark.

MarkCommand and UnmarkCommand are very similar in nature, sharing many
lines of duplicated codes.

Duplicated codes contributes to a longer code and possibly more issues
arising due to more lines of code. Review process will be increasingly
tedious as well.

Let's abstract the functionality of Mark and Unmark to ChangeMark in
TaskList, Storage and Ui.

Commit:
Edited checkstyle for ChangeMark.

Commit:
Merge pull request #3 from jltham/branch-C-Update

Created new classes: ChangeMark to contain both Mark and Unmark.

Commit:
Added update and clone functionalities.

Commit:
Merge pull request #4 from jltham/branch-C-Update

Added update and clone functionalities.

Commit:
Merge branch 'branch-C-Update'

Commit:
Merge branch 'master' of https://github.com/jltham/ip

* 'master' of https://github.com/jltham/ip:

Commit:
Added profile pictures

Commit:
Added User Guide and updated some functionalities.

Commit:
Added a new test for Tasks

Commit:
Set theme jekyll-theme-slate

Commit:
Set theme jekyll-theme-minimal

Commit:
Set theme jekyll-theme-minimal

Commit:
Set theme jekyll-theme-tactile

Commit:
Fix code quality feedback

Commit:
Merge branch 'master' of https://github.com/jltham/ip

Commit:
Update Duke based on comments received.

Commit:
Enhance functionalities with more checks.

Corrected file path for storage file creation.
